integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/HelloSquoosh.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose.testapp.validation.examples
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import com.android.designcompose.DesignDocSettings
+import com.android.designcompose.LocalDesignDocSettings
+
+@Composable
+fun HelloSquoosh() {
+    CompositionLocalProvider(LocalDesignDocSettings provides DesignDocSettings(useSquoosh = true)) {
```

Comment:
Sorry, I think I gave bad feedback. This should look like:

```kotlin
DesignDocSettings(useSquoosh = true) {
  HelloWorldDoc.Main(name = "Squoosh")
}
```

Reply:
I think, anyway, it was the "With" thing I wasn't sure about. Perhaps other CompositionLocal users provide a "With" composable.

Reply:
I just came to the conclusion that it was better to be more explicit about the mechanism that we were using, rather than hiding the implementation detail. This gives the user more flexibility to change any of the settings (like a docOverride?) at runtime. [Material's docs ](https://developer.android.com/develop/ui/compose/designsystems/material#emphasis) directly use CompositionLocal so I don't think it should be an issue for us to use it here.

Reply:
Oh, cool -- doing what Material does is good for me! Thanks for finding this, Guru!